### `Quantis e Quartis`

### Ajudam a entender o que ocorre em certa porcentagem dos nossos dados. Eles recortam nossos dados na quantidade que pedimos, e então temos uma noção melhor de como as observações estão se comportando.

<br>

![alt text](https://academy.vertabelo.com/static/python-data-science-04-quartiles-3.png)

<br>

### O percentil 10% (quantil 0.10), por exemplo, nos informa qual é o valor máximo dos dados se eu considerar somente os 10% primeiros termos dos meus dados, depois de ordená-los.

## Os quartis são quantis específicos:

#### * 1˚ Quartil = Quantil 0.25
#### * 2˚ Quartil = Mediana = Quantil 0.50
#### * 3˚ Quartil = Quantil 0.75

<br>

### Usando os dados do preço do Big Mac de Julho de 2022.

<br>

https://www.kaggle.com/datasets/vittoriogiatti/bigmacprice

In [14]:
import pandas as pd

In [20]:
df = pd.read_csv('BigmacPrice.csv')

In [21]:
df_filtered = df[df['date'] == '2022-07-01']

In [22]:
df_filtered.head()

date currency_code        name  local_price  dollar_ex  \
1876  2022-07-01           ARS   Argentina       590.00        129   
1877  2022-07-01           AUD   Australia         6.70          1   
1878  2022-07-01           EUR     Austria         4.35          1   
1879  2022-07-01           AZN  Azerbaijan         4.70          1   
1880  2022-07-01           BHD     Bahrain         1.60          1   

      dollar_price  
1876          4.57  
1877          6.70  
1878          4.35  
1879          4.70  
1880          1.60

In [23]:
df_filtered.count()

date             70
currency_code    70
name             70
local_price      70
dollar_ex        70
dollar_price     70
dtype: int64

### Vamos calcular alguns quantis para o  preço do Big Mac em dólares americanos

In [25]:
df_filtered['dollar_price'].quantile(0.1) ### Aqui calculamos o quantil 0.1, ou o percentil 10%

2.413

Podemos afirmar que 10% dos Big Mac's do mundo custam até 2.413 dólares

In [29]:
df_filtered['dollar_price'].quantile(0.25) ### Aqui calculamos o quantil 0.25, ou o percentil 25%, ou o primeiro quartil

3.0

Podemos afirmar que um quarto dos Big Mac's do mundo custam até 3 dólares

In [30]:
df_filtered['dollar_price'].quantile(0.5) ### Aqui calculamos o quantil 0.5, ou o percentil 50%, ou a mediana

3.9050000000000002

In [31]:
df_filtered['dollar_price'].median()

3.9050000000000002

Podemos dizer que metade dos Big Mac's do mundo custam até 3.9 dólares


### Agora com os quantis, conseguimos montar um retrato mais completo de como nossos dados se comportam.

In [33]:
percentil_10 = df_filtered['dollar_price'].quantile(0.1)

percentil_25 = df_filtered['dollar_price'].quantile(0.25)

percentil_50 = df_filtered['dollar_price'].quantile(0.50)

percentil_75 = df_filtered['dollar_price'].quantile(0.75)

percentil_90 = df_filtered['dollar_price'].quantile(0.9)

percentil_95 = df_filtered['dollar_price'].quantile(0.95)

percentil_99 = df_filtered['dollar_price'].quantile(0.99)

media = df_filtered['dollar_price'].mean()

moda = df_filtered['dollar_price'].mode()[0]

In [34]:
print("Quantil 0.10: {}\n".format(percentil_10))

print("Primeiro Quartil: {}\n".format(percentil_25))

print("Mediana: {}\n".format(percentil_50))

print("Média: {}\n".format(media))

print("Moda: {}\n".format(moda))

print("Terceiro Quartil: {}\n".format(percentil_75))

print("Quantil 0.90: {}\n".format(percentil_90))

print("Quantil 0.95: {}\n".format(percentil_95))

print("Quantil 0.99: {}\n".format(percentil_99))

Quantil 0.10: 2.413

Primeiro Quartil: 3.0

Mediana: 3.9050000000000002

Média: 4.003571428571429

Moda: 4.0

Terceiro Quartil: 4.6875

Quantil 0.90: 5.91

Quantil 0.95: 6.609999999999999

Quantil 0.99: 6.9551



### Os quantis são de muita importância quando queremos entender melhor a própria dispersão dos dados que estamos analisando. Alguns exemplos dentre as várias aplicações para eles:

* Identificar comportamentos muito diferentes do restante dos dados (Outliers)
* Entender em qual corte a maioria dos dados se encontram, ou até mesmo uma parte específica deles
* Ajudar na tomada de decisão em situações de Trade-Off. Ex: Acurácia de Localização
* Boxplot

Na empolgante jornada do Data Science, os **quantis** e **quartis** despontam como ferramentas valiosas para desvendar os segredos da distribuição dos dados. Eles fornecem insights poderosos sobre como os valores se dividem, revelando padrões e assimetrias que podem ser cruciais para diversas análises.

<br>

**Mergulhando no Universo dos Quantis:**

Imagine um conjunto de dados como uma fila de espera em um banco. Os quantis nos dividem em grupos, como se fossem guichês de atendimento. Cada guichê representa um quantil, com um número específico de pessoas esperando.

* **Quantil 25% (Q1):** As primeiras 25% das pessoas da fila já estão sendo atendidas.
* **Quantil 50% (Mediana):** 50% das pessoas já foram atendidas, este é o ponto central da fila.
* **Quantil 75% (Q3):** 75% das pessoas já foram atendidas, o que significa que os 25% restantes ainda aguardam.

**Desvendando os Segredos dos Quantis:**

Os quantis nos revelam muito mais do que apenas dividir os dados em grupos. Eles nos informam sobre a **posição** de cada valor dentro da distribuição.

* **Valores próximos ao Q1:** Indicam que estão abaixo de 25% dos dados, ou seja, são menores que a maioria dos valores.
* **Valores próximos à Mediana:** Indicam que estão próximos do "centro" dos dados, ou seja, são valores medianos.
* **Valores próximos ao Q3:** Indicam que estão acima de 75% dos dados, ou seja, são maiores que a maioria dos valores.

<br>

**Exemplos para Dominar a Linguagem dos Quantis:**

**Exemplo 1: Salários de Funcionários:**

* **Q1:** R$ 2.000,00

* **Mediana (Q2):** R$ 3.500,00

* **Q3:** R$ 5.000,00

* **Interpretação:**
    * 25% dos funcionários ganham menos de R$ 2.000,00.
    * 50% dos funcionários ganham entre R$ 2.000,00 e R$ 5.000,00.
    * 25% dos funcionários ganham mais de R$ 5.000,00.

<br>

**Exemplo 2: Tempo de Espera em Fila:**

* **Q1:** 5 minutos
* **Mediana:** 10 minutos
* **Q3:** 15 minutos
* **Interpretação:**
    * 25% das pessoas esperam menos de 5 minutos.
    * 50% das pessoas esperam entre 5 e 15 minutos.
    * 25% das pessoas esperam mais de 15 minutos.

<br>

**Quantis para Dominar Data Science:**

Os quantis se tornam campeões em Data Science em diversas situações:

* **Análise da Distribuição de Dados:** Identificam assimetrias e pontos de inflexão na distribuição dos dados, auxiliando na compreensão do comportamento geral dos dados.
* **Detecção de Anomalias:** Permitem identificar valores atípicos (outliers) que podem distorcer a análise.
* **Comparação de Conjuntos de Dados:** Facilitam a comparação da distribuição de dados de diferentes origens, mesmo com escalas diferentes.
* **Desenvolvimento de Modelos Preditivos:** Auxiliam na construção de modelos preditivos mais robustos e precisos.

**Dominando os Quantis: Uma Jornada Essencial para Data Scientists:**

Ao dominar os quantis, Data Scientists ganham chaves valiosas para desvendar os segredos da distribuição dos dados, extraindo insights poderosos que guiam decisões mais assertivas em diversos campos.

Lembre-se: a jornada de dominar os quantis é essencial para se tornar um mestre em Data Science!

<br>

**Quartis: O Quarteto Fantástico dos Quantis**

Os **quartis** são um caso especial dos quantis, dividindo os dados em **quatro partes iguais** (cada parte com 25% dos dados). Eles são representados por:

* **Primeiro quartil (Q1):** Divide os dados em 25% menores e 75% maiores.
* **Segundo quartil (Mediana):** Divide os dados em 50% menores e 50% maiores.
* **Terceiro quartil (Q3):** Divide os dados em 75% menores e 25% maiores.

Os quartis forne